src/ir_def/ir.ml

```diff
@@ -134,7 +134,7 @@ and prim =
   | IdxBlobPrim                       (* blob indexing *)
   | BreakPrim of id                   (* break *)
   | RetPrim                           (* return *)
-  | AwaitPrim of Type.async_sort       (* await/await* *)
+  | AwaitPrim of await_sort           (* await/await* *)
```

Comment:
```suggestion
  | AwaitPrim of Type.async_sort * Type.async_sort           (* await/await* *)
```
Might be cleaner. And then you rule out the case where the pair (T.Fut, T.Cmp) is in check_ir and `short-circuit` when the argument is (T.Cmp, T.Fut).

Reply:
I'll do a follow-up PR for this cleanup.